src/grammar/__snapshots__/grammar.spec.ts.snap

```diff
@@ -182,6 +182,54 @@ Line 2, col 15:
 "
 `;
 
+exports[`grammar should fail case-19 1`] = `
+"<unknown>:1:14: Empty parameter list should not have a dangling comma.
+Line 1, col 14:
+> 1 | fun function(,) {}
+                   ^
+  2 | 
+"
+`;
+
+exports[`grammar should fail case-20 1`] = `
+"<unknown>:2:24: Empty parameter list should not have a dangling comma.
+Line 2, col 24:
+  1 | fun another() {
+> 2 |     return 42.toString(,);
+                             ^
+  3 | }
+"
+`;
+
+exports[`grammar should fail case-21 1`] = `
+"<unknown>:1:14: Empty parameter list should not have a dangling comma.
+Line 1, col 14:
+> 1 | fun function(,) : Int {
+                   ^
+  2 |     return 42;
+"
+`;
+
+exports[`grammar should fail case-22 1`] = `
+"<unknown>:6:14: Empty parameter list should not have a dangling comma.
+Line 6, col 14:
+  5 | fun b(): Int {
+> 6 |     return a(,);
+                   ^
+  7 | }
+"
+`;
+
+exports[`grammar should fail case-23 1`] = `
+"<unknown>:7:31: Empty parameter list should not have a dangling comma.
+Line 7, col 31:
+  6 | 
+> 7 | abstract fun testFuncAbstract(,);
```

Comment:
I only see one error here, but the `test-failed/case-23` contains several of them

Reply:
Definitely doing small test files next time to avoid missing that :-)